cmd/habitat/community/consensus/raft/service.go

```diff
@@ -0,0 +1,289 @@
+package raft
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/pkg/raft/fsm"
+	"github.com/eagraf/habitat/pkg/raft/transport"
+	"github.com/hashicorp/raft"
+	raftboltdb "github.com/hashicorp/raft-boltdb"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	RetainSnapshotCount = 1000
+	RaftTimeout         = 10 * time.Second
+)
+
+// ClusterService is an implementation of cluster.ClusterManager
+type ClusterService struct {
+	mux       transport.RaftMultiplexer
+	instances map[string]*raftClusterInstance
+
+	nodeID string
+}
+
+type raftClusterInstance struct {
+	communityID  string
+	serverID     string
+	address      string
+	instance     *raft.Raft
+	stateMachine *fsm.CommunityStateMachine
+}
+
+func NewClusterService() *ClusterService {
+	cs := &ClusterService{
+		mux:       *transport.NewRaftMultiplexer(),
+		instances: make(map[string]*raftClusterInstance),
+
+		nodeID: compass.NodeID(),
+	}
+
+	return cs
+}
+
+func (cs *ClusterService) Start() error {
+	go cs.mux.Listen(getMultiplexerAddress())
+	return nil
+}
+
+func (cs *ClusterService) CreateCluster(communityID string) error {
+	if _, ok := cs.instances[communityID]; ok {
+		return fmt.Errorf("raft instance for community %s already initialized", communityID)
+	}
+
+	stateMachine := fsm.NewCommunityStateMachine()
+
+	ra, ht, err := setupRaftInstance(communityID, stateMachine, true)
+	if err != nil {
+		return fmt.Errorf("failed to setup raft instance: %s", err.Error())
+	}
+
+	err = cs.mux.RegisterListener(communityID, ht)
+	if err != nil {
+		return fmt.Errorf("failed to register listener: %s", err.Error())
+	}
+
+	raftInstance := &raftClusterInstance{
+		communityID:  communityID,
+		serverID:     getServerID(communityID),
+		address:      getCommunityExternalAddress(communityID),
+		instance:     ra,
+		stateMachine: stateMachine,
+	}
+	cs.instances[communityID] = raftInstance
+
+	return nil
+}
+
+func (cs *ClusterService) RemoveCluster(communityID string) error {
+	return nil
```

Comment:
what is this supposed to do?

Reply:
This should be called if someone leaves a community, I forgot to implement it

Reply:
Same goes for RemoveNode